### PR TITLE
Remove defer chain logic

### PR DESCRIPTION
**Pull request by ianrowan**
## Link to Issue
Closes: #4136 

## Description of Changes
- Removes Defer Chain logic per discovery chain was only loaded where needed 

## Test Plan
- Go to token communities w/ on-chain actions and ensure proposals, etc pages load
- Can flag `initChain/Data` and `initAPI` with console logs to ensure these are not loaded on Threads or any other pages

### CODE REVIEW

File: packages/commonwealth/client/scripts/helpers/chain.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export const deinitChainOrCommunity = async () => {
 // initChain fn ought to proceed or abort
 export const selectChain = async (
   chain?: ChainInfo,
-  deferred = false
+  useChain = false
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/helpers/chain.ts:31]: Can we remove this variable? @ianrowan

**Comment by ianrowan** [comment]
Found issue with chain not loaded when clicking on a proposal link without having visited proposal/governance page. Should be easy fix to mirror logic in proposals

**Review by jnaviask — CHANGES_REQUESTED** [review]
- Getting double-initialization for chain when refreshing a dydx proposal page (was using /dydx/proposal/12). Might need to use similar `useRef` logic as for cosmos proposal page. Same on edgeware. Breaks loading on dydx with Not Found.
- Not related to your changes but getting a lot of 401s on /getLinks

**Comment by ianrowan** [comment]
> * Getting double-initialization for chain when refreshing a dydx proposal page (was using /dydx/proposal/12). Might need to use similar `useRef` logic as for cosmos proposal page. Same on edgeware. Breaks loading on dydx with Not Found.
> * Not related to your changes but getting a lot of 401s on /getLinks

Added the same logic as cosmos proposal page and double-init seems to be fixed. 

As for 401s havent reproduced anywhere but could address separately 
@jnaviask

**Review by jnaviask — APPROVED** [review]

**Pull request merged**